### ISSUE TITLE
feature request: Improved resize / placement support

### ISSUE THREAD

**Issue by tones111** [google/gxui#54]
I'm finding it difficult to mock up a ui that handles window resizing or allowing fill space between components.  For example, there doesn't seem to be a mechanism to have some components at the top of the window, with a status bar across the bottom.  Enabling the layout border to be drawn it can be seen that the layout size is not adapting to changes in the window size.  Attempting to fill space with an invisible component (ex: a label with no text) and a large maximum desired size doesn't seem to absorb the extra space either.

**Comment by ben-clayton** [comment]
> Enabling the layout border to be drawn it can be seen that the layout size is not adapting to changes in the window size. 

I assume you're referring to [LinearLayout](https://github.com/google/gxui/blob/master/linear_layout.go). If so, this layout is [really simple / dumb](https://github.com/google/gxui/blob/master/mixins/parts/linear_layout.go#L37) - it will start from either the top (for vertical orientation) or left (for horizontal orientation), and layout each of the children to their desired size. The desired size is always limited to the remaining space, which decreases with each consecutive child layout.

You are completely right in that this is not perfectly achievable at the moment. @mattn stumbled across a similar limitation when implementing his GXUI based [twitter client](https://github.com/mattn/gxuitter). In his case, he was after the textbox and send button at the bottom of the window, and a list that filled all remaining space above. In this case, we could solve his problem by [adding LinearLayout support for bottom-to-top and right-to-left layouts](https://github.com/google/gxui/issues/21). The interim [solution I offered was to use a SplitterLayout](https://github.com/mattn/gxuitter/issues/1#issuecomment-83618360) - this isn't perfect either as the split is based on a ratio of weights - there's no way to be explicit about the split in precise units.

Basically, we're lacking more advanced layout containers. I suspect that the LinearLayout [improvements](https://github.com/google/gxui/issues/21) will help provide a solution to a lot of the impossible cases, but relative and absolute layout containers would also be wanted.

If anyone wants to help design and implement these, they would be greatly appreciated.

**Comment by phaikawl** [comment]
Is it a good idea to model the layouts after some other popular libraries? We could use Qt, Swing, GTK, .NET or something as a base for our design. I don't know which one has the best layout model, but we could do an investigation. I think we could save a lot of time and energy by doing that.
